doc/gen_support_matrix_html.rs

```diff
@@ -321,25 +336,45 @@ fn render_html(matrix: &schema::Matrix, board_support_tier: &SupportTier) -> Res
                     icon: support_key.icon.to_owned(),
                     description: support_key.description.to_owned(),
                 })
-            })
-            .collect::<Result<Vec<_>, Error>>()?;
-
-
-        let chip = matrix.chips.get(&board_info.chip).ok_or(Error::InvalidChipName {
-            found: board_info.chip.to_owned(),
-            board: board_name.to_owned(),
-        })?;
-
-        Ok(BoardSupport {
-            chip: chip.name.to_owned(),
-            chip_technical_name: board_info.chip.to_owned(),
-            url: board_info.url.to_owned(),
-            technical_name: board_technical_name.to_owned(),
-            name: board_name.to_owned(),
-            tier: board_info.tier.to_owned(),
-            functionalities,
-        })
-    }).collect::<Result<Vec<_>, Error>>()?;
+            });
+            let errors = functionalities
+                .clone()
+                .filter_map(|f| f.err())
+                .collect::<Vec<_>>();
+            if errors.is_empty() {
+                Ok(BoardSupport {
+                    chip: chip_info.name.to_owned(),
+                    chip_technical_name: board_info.chip.to_owned(),
+                    url: board_info.url.to_owned(),
+                    technical_name: board_technical_name.to_owned(),
+                    name: board_name.to_owned(),
+                    tier: board_info.tier.to_owned(),
+                    functionalities: functionalities.map(|f| f.unwrap()).collect(),
+                })
+            } else {
+                Err(errors)
+            }
+        });
+
+    let errors = boards
+        .clone()
+        .filter_map(|f| f.err())
+        .flatten()
+        .collect::<Vec<_>>();
+    if errors.is_empty() {
+        Ok(boards.map(|f| f.unwrap()).collect())
+    } else {
+        Err(Error::ValidationIssues { errors: errors })
```

Comment:
optional: shorter and more idiomatic, otherwise it looks like there could be a typo: 
```suggestion
        Err(Error::ValidationIssues { errors })
```